spec/support/global_state_reset.rb

```diff
@@ -0,0 +1,22 @@
+module OpenProject
+  ##
+  # Provides resets to any global state that may have been changed through tests
+  # and the change of which should not affect other tests.
+  module GlobalStateReset
+    module_function
+
+    def reset_global_state!
+      reset_current_user!
+    end
+
+    def reset_current_user!
+      User.current = User.anonymous
+    end
+  end
+end
+
+RSpec.configure do |config|
+  config.after(:suite) do
+    OpenProject::GlobalStateReset.reset_global_state!
```

Comment:
@ulferts you simplified the global state cleanup on the Rails 4 branch: https://github.com/opf/openproject/commit/581a547e5554fba150a57c8d2bade071439b2c21 would that patch help here?